dao/src/main/java/com/blog/dao/jdbc/PostDaoImpl.java

```diff
@@ -116,40 +114,38 @@ public Post getPostById(Long id) throws DataAccessException {
     public List<Post> getAllPostsByAuthorId(Long authorId) throws DataAccessException {
         LOGGER.debug("Get list of posts by author id = [{}] from database.", authorId);
         SqlParameterSource parameterSource = new MapSqlParameterSource(AUTHOR_ID, authorId);
-        return jdbcTemplate.query(getAllPostsByAuthorIdSql, parameterSource, (resultSet, i) -> new PostShortRowMapper().mapRow(resultSet, i));
+        return jdbcTemplate.query(getAllPostsByAuthorIdSql, parameterSource, (resultSet, i) -> postShortRowMapper.mapRow(resultSet, i));
     }
 
     public List<Post> getPostsByInitialIdAndQuantity(Long initial, Long quantity) throws DataAccessException {
         LOGGER.debug("Get list of posts by initial = [{}] and quantity = [{}] from database.", initial, quantity);
-        MapSqlParameterSource parameterSource = getParameterSourceForItintalAndQuantity(initial, quantity);
-        return jdbcTemplate.query(getAllPostsByInitialIdAndQuantitySql, parameterSource, (resultSet, i) -> new PostShortRowMapper().mapRow(resultSet, i));
+        MapSqlParameterSource parameterSource = getParameterSourceForInitialAndQuantity(initial, quantity);
+        return jdbcTemplate.query(getAllPostsByInitialIdAndQuantitySql, parameterSource, (resultSet, i) -> postShortRowMapper.mapRow(resultSet, i));
     }
 
-    @Override
     public List<Post> getPostsByInitialIdAndQuantity(Long initial, Long quantity, String sort) throws DataAccessException {
         LOGGER.debug("Get list of posts by initial = [{}], quantity = [{}] and sort = [{}] from database.", initial, quantity, sort);
-        MapSqlParameterSource parameterSource = getParameterSourceForItintalAndQuantity(initial, quantity);
-        parameterSource.addValue(SORT, sort);
-        return jdbcTemplate.query(getAllPostsByInitialIdAndQuantitySql, parameterSource, (resultSet, i) -> new PostShortRowMapper().mapRow(resultSet, i));    }
+        MapSqlParameterSource parameterSource = getParameterSourceForInitialAndQuantity(initial, quantity);
+        return jdbcTemplate.query(getAllPostsByInitialIdQuantityAndSortSql + " " + sort + " DESC LIMIT :initial, :quantity", parameterSource, (resultSet, i) -> postShortRowMapper.mapRow(resultSet, i));
```

Comment:
Think it would be better to use String.format:
`String.format( "Select * from blabla where id = :id %s DESC LIMIT :initial, :quantity", sort)`